nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/FlowResource.java

```diff
@@ -2057,6 +2062,100 @@ public Response getDetails(
         return generateOkResponse(flowDetails).build();
     }
 
+    @GET
+    @Consumes(MediaType.WILDCARD)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("registries/{registry-id}/branches/{branch-id-a}/buckets/{bucket-id-a}/flows/{flow-id-a}/{version-a}/diff/branches/{branch-id-b}/buckets/{bucket-id-b}/flows/{flow-id-b}/{version-b}")
```

Comment:
Is the intended use case for this to right-click on a process group under version control and be able to choose something like `Compare With` and then choose another version to compare with? is there any other scenario where we want to compare two flow versions?

I'm asking because if it is limited to the first scenario I described, we may want to consider a more specific API to the PG, similar to `process-groups/{id}local-changes`, maybe `/process-groups/{id}/diff`. Then it would only need the parameters for the other flow version to compare to, which could be added to the path, or as query params.

Reply:
The main use case is to provide a possibility to make comparisons even _before_ picking a version. As for the user's perspective this will result a similar listing as we have with `Local changes`. The intended place for this is the version list (for example when exeucting `Change Version` action on a versioned group as an addition.

The URL deliberately consists this many parameteres: the simplest situation is where we want to compers consecurtive versions of the same flow (in the same branch, etc.). A more complex, but still useful scenario is to compare flow snapshots between branches, like between a production and a development branch. As the branching strategy does not provide rescritions to follow the same structure for every branch, stricting this might make it impossible to make useful comparions with certain setups.

Reply:
So just to clarify, you are saying that there is some PG under version control, lets say it is at v1. The user can select Change Version and see a versions list which has (1, 2, 3), and they can somehow select 2 and 3 and compare those even though the current flow is at 1? or can they only compare 1 against 2 or 3?

Reply:
As for URL perspective, other then the registry id, it is expected to provide branch-bucket-flow-version information for both sides of the comparson.

As for the user interface, the idea is to have a button with every version, like `Compare To...` which leads to the comparison view, which should be similart to the `Local Changes`, expect additional controls to pick the information above. The left side of the comparison should be already filled based on the picked version.